src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -618,6 +618,44 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
         })
     }
 
+    /// Convert a PrimitiveType to a Ty, where possible.
+    ///
+    /// This is used for resolving trait impls for primitives
+    fn primitive_type_to_ty(&mut self, prim: PrimitiveType) -> Option<Ty<'tcx>> {
+        use PrimitiveType::*;
+        let tcx = self.cx.tcx;
+
+        Some(tcx.mk_ty(match prim {
+            Bool => ty::Bool,
+            Str => ty::Str,
+            Char => ty::Char,
+            Never => ty::Never,
+            I8 => ty::Int(ty::IntTy::I8),
+            I16 => ty::Int(ty::IntTy::I16),
+            I32 => ty::Int(ty::IntTy::I32),
+            I64 => ty::Int(ty::IntTy::I64),
+            I128 => ty::Int(ty::IntTy::I128),
+            Isize => ty::Int(ty::IntTy::Isize),
+            F32 => ty::Float(ty::FloatTy::F32),
+            F64 => ty::Float(ty::FloatTy::F64),
+            U8 => ty::Uint(ty::UintTy::U8),
+            U16 => ty::Uint(ty::UintTy::U16),
+            U32 => ty::Uint(ty::UintTy::U32),
+            U64 => ty::Uint(ty::UintTy::U64),
+            U128 => ty::Uint(ty::UintTy::U128),
+            Usize => ty::Uint(ty::UintTy::Usize),
+            //ty::Tuple(tys) if tys.is_empty() => Res::Primitive(Unit),
+            //ty::Tuple(_) => Res::Primitive(Tuple),
+            //ty::Array(..) => Res::Primitive(Array),
+            //ty::Slice(_) => Res::Primitive(Slice),
+            //ty::RawPtr(_) => Res::Primitive(RawPointer),
+            //ty::Ref(..) => Res::Primitive(Reference),
+            //ty::FnDef(..) => panic!("type alias to a function definition"),
+            //ty::FnPtr(_) => Res::Primitive(Fn),
```

Comment:
So these commented out lines can't be converted, if I understand correctly. The `PrimitiveType` enum does not have any parameters for these types, whereas the `ty` counterpart requires more information.

Let me know if I'm mistaken here and if so, any guidance on how to convert them would be appreciated. Otherwise, I'll just delete these comments.

```suggestion
```

Reply:
You could maybe instantiate the types with fresh generic parameters? Not sure how though.

Reply:
I think I would need actual information, e.g. for [`ty::Array`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/enum.TyKind.html#variant.Array) we have:

```
Array(Ty<'tcx>, &'tcx Const<'tcx>)
Tuple Fields
0: Ty<'tcx>
1: &'tcx Const<'tcx>

An array with the given length. Written as [T; n].
```

Which means I need to know the type and size of the array.

Reply:
I _suspect_ you can generate a type with generic parameters but I'm not sure how best to do that here. Fine to file a followup

Reply:
Okay I will create a follow-up issue for this, perhaps it can be addressed somehow

Reply:
Added a comment to the original issue, I think that issue covers these cases, so by this PR it will be "partially" fixed. Or would you rather I create a new separate issue?

https://github.com/rust-lang/rust/issues/90703#issuecomment-1004263455

Reply:
Can you open a separate issue? That issue will be closed by this PR and having a separate issue will make it easier to track.

Reply:
https://github.com/rust-lang/rust/issues/92536